src/google/adk/agents/loop_agent.py

```diff
@@ -48,9 +48,12 @@ async def _run_async_impl(
     while not self.max_iterations or times_looped < self.max_iterations:
       for sub_agent in self.sub_agents:
         async for event in sub_agent.run_async(ctx):
-          yield event
           if event.actions.escalate:
+            # Reset the escalate status to make sure this escalation only exits one level of the loop agent, not the entire nested loop agents structure.
+            event.actions.escalate = False
```

Comment:
We should instead check whether this is no more loop agent between this agent and the event author agent. 

---

`event.actions.escalate` is the conversation history and we shouldn't alter it.